### PR TITLE
Do not report CA2235 for static and const fields

### LINKED ISSUE #1883: CA2235 raised for static and const fields
#### Analyzer package

[Microsoft.NetFramework.Analyzers](https://www.nuget.org/packages/Microsoft.NetFramework.Analyzers/)

#### Package Version

v2.6.2

#### Diagnostic ID

[CA2235](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2235-mark-all-non-serializable-fields)

#### Repro steps

Class with attribute serializable and with const or static fields.

#### Expected behavior
No diagnostic warning should be raised because this are not field instances.

#### Actual behavior
Raises diagnostic warning.

### PR DESCRIPTION
**Pull request by mavasani**
Diagnostic is only intended for instance fields.
Fixes #1883

### CODE REVIEW

File: src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs [hunk 1]
```diff
@@ -224,6 +224,12 @@ public void AnalyzeSymbol(SymbolAnalysisContext context)
                         namedTypeSymbol.GetMembers().OfType<IFieldSymbol>().Where(m => !IsSerializable(m.Type));
                     foreach (IFieldSymbol field in nonSerializableFields)
                     {
+                        // Only process instance fields
+                        if (field.IsConst || field.IsStatic)
```
Comment (sharwell) [inline src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs]: ❔ Does `IsConst` not imply `IsStatic`? In other words, is it sufficient to use the following:

```csharp
if (field.IsStatic)
{
  continue;
}
```

File: src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs [hunk 2]
```diff
@@ -224,6 +224,12 @@ public void AnalyzeSymbol(SymbolAnalysisContext context)
                         namedTypeSymbol.GetMembers().OfType<IFieldSymbol>().Where(m => !IsSerializable(m.Type));
                     foreach (IFieldSymbol field in nonSerializableFields)
                     {
+                        // Only process instance fields
```
Comment (sharwell) [inline src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs]: 📝 I normally place comments like this immediately inside the braces. This allows *consistent* placement of documenting comments across `if` statements with a variety of `if`/`else if`/`else` forms.

File: src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs (hunk 1 above)
Comment (mavasani) [reply src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs]: Yep, changed.

File: src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs (hunk 2 above)
Comment (mavasani) [reply src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs]: I followed the pattern used throughout this analyzer.

**Pull request merged**